tests/integ/test_local_mode.py

```diff
@@ -230,6 +251,53 @@ def test_tf_local_data_local_script():
         fcntl.lockf(local_mode_lock, fcntl.LOCK_UN)
 
 
+def test_local_mode_serving_from_s3_model(sagemaker_local_session, mxnet_model):
+    local_mode_lock_fd = open(LOCK_PATH, 'w')
+    local_mode_lock = local_mode_lock_fd.fileno()
+
+    model_data = mxnet_model.model_data
+    boto_session = sagemaker_local_session.boto_session
+    default_bucket = sagemaker_local_session.default_bucket()
+    uploaded_data = tar_and_upload_dir(boto_session, default_bucket,
+                                       'test_mxnet_local_mode', '', model_data)
+
+    s3_model = MXNetModel(model_data=uploaded_data.s3_prefix, role='SageMakerRole',
+                          entry_point=mxnet_model.entry_point, image=mxnet_model.image,
+                          sagemaker_session=sagemaker_local_session)
+
+    try:
+        # Since Local Mode uses the same port for serving, we need a lock in order
+        # to allow concurrent test execution. The serving test is really fast so it still
+        # makes sense to allow this behavior.
+        fcntl.lockf(local_mode_lock, fcntl.LOCK_EX)
+        predictor = s3_model.deploy(initial_instance_count=1, instance_type='local')
+        data = numpy.zeros(shape=(1, 1, 28, 28))
+        predictor.predict(data)
+    finally:
+        mxnet_model.sagemaker_session.delete_endpoint('')
```

Comment:
Is the empty string passed here because that's the name we explicitly chose for the endpoint? 

Can we add delete_endpoint to the predictor class to make this more intuitive? That's where it should be anyway, IMO.